pallets/exchange/src/lib.rs

```diff
@@ -181,8 +181,14 @@ pub mod pallet {
 		/// Insufficient asset balance.
 		InsufficientAssetBalance,
 
-		/// Given trading limit has been exceeded (Sell) or has Not been reached (buy).
-		AssetBalanceLimitExceeded,
+		/// Given trading limit has been exceeded (buy).
+		TradeAmountExceededLimit,
+
+		/// Given trading limit has not been reached (sell).
+		TradeAmountNotReachedLimit,
+
+		/// Intentions for direct trading are matched wrong.
+		IncorrectIntentionsMatch,
```

Comment:
If I understand the scenario right, when this error occurs it means that main intention and array of matched intentions was choosen incorrectly as parameters of `resolve_matched_intentions` func. That's why i picked up this name.